R/calculate_within_batch_ari.R

```diff
@@ -0,0 +1,100 @@
+#' Calculate within-batch ARI scores from a merged SCE object.
+#'
+#'
+#' @param individual_sce_list A list of individual SCE objects
+#' @param pc_name The name that allows access to the PCs in the merged SCE
+#'  object. Example: "fastMNN_PCA".
+#' @param merged_sce The merged SCE object containing data from multiple batches
+#' @param batch_column The variable in `merged_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "library_id".
+#' @param seed Seed for initializing random sampling
+#'
+#' @return Tibble with three columns: `ari`, representing the calculated ARI values;
+#'   `batch_id`, the batch id associated with each `ari`; `pc_name`, the name
+#'   associated with the pc results
+#'
+#' @import SingleCellExperiment
+#' @importFrom rlang .data
+#'
+#' @export
+calculate_within_batch_ari <- function(individual_sce_list,
+                                       pc_name,
+                                       merged_sce,
+                                       batch_column = "library_id",
+                                       seed = NULL) {
+  # Set the seed for subsampling and clustering
+  set.seed(seed)
+
+  # Check that `batch_column` is in colData of SCE
+  if (!batch_column %in% colnames(colData(merged_sce))) {
+    stop("The specified batch column is missing from the colData of the SingleCellExperiment object.")
```

Comment:
```suggestion
    stop("The specified batch column is missing from the colData of the `merged_sce`.")
```
Since there are two objects being input, let's be specific.